sdk/src/main/scala/com/horizen/block/MainchainTxSidechainCreationCrosschainOutput.scala

```diff
@@ -59,7 +49,32 @@ object MainchainTxSidechainCreationCrosschainOutput {
     //val certVk: Array[Byte] = sidechainCreationOutputBytes.slice(currentOffset, currentOffset + certVkSize)
     //currentOffset += certVkSize
 
-    new MainchainTxSidechainCreationCrosschainOutput(sidechainCreationOutputBytes.slice(offset, currentOffset),
-      sidechainId, withdrawalEpochLength, amount, address, customData, constant, certVk)
+    new MainchainTxSidechainCreationCrosschainOutputData(sidechainCreationOutputBytes.slice(offset, currentOffset),
+      withdrawalEpochLength, amount, address, customData, constant, certVk)
   }
 }
+
+class MainchainTxSidechainCreationCrosschainOutput(override val sidechainId: Array[Byte], data: MainchainTxSidechainCreationCrosschainOutputData)
+                                                    extends MainchainTxSidechainCreationCrosschainOutputData(
+                                                      data.sidechainCreationOutputBytes,
+                                                      data.withdrawalEpochLength,
+                                                      data.amount,
+                                                      data.address,
+                                                      data.customData,
+                                                      data.constant,
+                                                      data.certVk
+                                                    ) with MainchainTxCrosschainOutput {
+  override val outputType: Byte = MainchainTxSidechainCreationCrosschainOutput.OUTPUT_TYPE
+
+  override lazy val hash: Array[Byte] = BytesUtils.reverseBytes(Utils.doubleSHA256Hash(sidechainCreationOutputBytes))
+}
+
+
+object MainchainTxSidechainCreationCrosschainOutput {
+  val OUTPUT_TYPE: Byte = 3.toByte
+
+  def calculateSidechainId(index: Int, transactionHash: Array[Byte]): Array[Byte] = {
```

Comment:
switch to `calculateSidechainId(transactionHash: Array[Byte], index: Int)` to have arguments in the same order how it's used below.

Reply:
done